splitio/engine/impressions/unique_keys_tracker.py

```diff
@@ -1,22 +1,46 @@
 import abc
 import threading
 import logging
+
 from splitio.engine.filters import BloomFilter
+from splitio.optional.loaders import asyncio
 
 _LOGGER = logging.getLogger(__name__)
 
-class BaseUniqueKeysTracker(object, metaclass=abc.ABCMeta):
-    """Unique Keys Tracker interface."""
+class UniqueKeysTrackerBase(object, metaclass=abc.ABCMeta):
+    """Unique Keys Tracker base class."""
 
     @abc.abstractmethod
     def track(self, key, feature_flag_name):
         """
         Return a boolean flag
-
         """
         pass
 
-class UniqueKeysTracker(BaseUniqueKeysTracker):
+    def set_queue_full_hook(self, hook):
+        """
+        Set a hook to be called when the queue is full.
+
+        :param h: Hook to be called when the queue is full
+        """
+        if callable(hook):
+            self._queue_full_hook = hook
+
+    def _add_or_update(self, feature_flag_name, key):
+        """
+        Add the feature_name+key to both bloom filter and dictionary.
+
+        :param feature_flag_name: feature flag name associated with the key
+        :type feature_flag_name: str
+        :param key: key to be added to MTK list
+        :type key: int
+        """
+        if feature_flag_name not in self._cache:
```

Comment:
Do you need a `with self._lock:` here?

Reply:
Not needed here since the calling line is within a self._lock block